components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/IdentityProviderManager.java

```diff
@@ -2079,6 +2108,11 @@ public IdentityProvider addIdPWithResourceId(IdentityProvider identityProvider,
         String resourceId = dao.addIdP(identityProvider, tenantId, tenantDomain);
         identityProvider = dao.getIdPByResourceId(resourceId, tenantId, tenantDomain);
 
+        // Validate whether the IdP satisfies the minimum requirements to be a JWT Issuer.
+        if (isJwtIssuer(identityProvider)) {
```

Comment:
Shouldn't we do this before the dao.addIdP(identityProvider, tenantId, tenantDomain) (#2108)

Instead of doing a separate DB call, we can set the required first class variable to indicate the IDP as a trusted JWT issuer and let the DAO layer handle persisting it as a property